### PR TITLE
[WFLY-18502][WFLY-18498] adds MATRIX_OS allowing customization of tes…

### PR DESCRIPTION
**Pull request by emmartins**
…t matrix OSes; adds support for optional quickstart-specific sh scripts to run before and after the tests"

### CODE REVIEW

File: .github/workflows/quickstart_ci.yml [hunk 1]
```diff
@@ -52,6 +66,16 @@ jobs:
           java-version: ${{ matrix.jdk }}
           distribution: 'temurin'
           cache: 'maven'
+      - name: Run before script
+        env:
+          FILE: "./quickstarts/.github/workflows/quickstart_${{ inputs.QUICKSTART_PATH }}_ci_before.sh"
```
Comment (kabir) [inline .github/workflows/quickstart_ci.yml:71]: Maybe but this one and the other file as the env for the whole job to reduce duplication? IMO having it at the top level also makes it more 'documented'

File: .github/workflows/quickstart_ci.yml [hunk 2]
```diff
@@ -93,11 +117,21 @@ jobs:
...
+              chmod +x $FILE 
+              bash $FILE
+          fi
+        shell: bash
       - uses: actions/upload-artifact@v3
         if: failure()
         with:
           name: surefire-reports-JDK${{ matrix.jdk }}-${{ matrix.os }}
-          path: 'quickstarts/$QUICKSTART_PATH/**/surefire-reports/*.txt'
+          path: 'quickstarts/${{ env.QUICKSTART_PATH }}/**/surefire-reports/*.txt'
```
Comment (kabir) [inline .github/workflows/quickstart_ci.yml:134]: I **_think_** this should be `${{ inputs.QUICKSTART_PATH }}` but am not 100% sure!

Comment (ehsavoie) [reply .github/workflows/quickstart_ci.yml:134]: +1

File: .github/workflows/quickstart_ci.yml (hunk 1 above)
Comment (emmartins) [reply .github/workflows/quickstart_ci.yml:71]: My idea was to have enforce each QS stays independent .yml and .sh wise. Duplication should be rare, if even will exist, and I will of course document this is at the google doc.

**Pull request merged**